### DISCUSSION TITLE
Is it correct way to access instance of application?

### DISCUSSION THREAD

**Discussion by saintserge** [mojolicious/mojo#1743] [Q&A] [answered]
Hi all,
I was trying to test my minion tasks, which in the end required to generate some templates. Before this code as in Controller and use render_to_string method.

So in the task I did next:
```
my $app   = Mojo::Server->new()->load_app( 'script/myapp' );
$output = $app->render_to_string( 'word/some_prepared_xml' ,%template_data );
```
In the app I have helper 

```
$self->helper(
		render_to_string => sub {
			my $c             = shift;
			my $template      = shift;
			my $template_data = shift;
			return $c->render_to_string( $template, %$template_data )
		} );
```

So far it is working, but slower because MyApp start from 0. 

But I think it is not totally proper way? 
If it is, can you please advice on how it can be done? Basically I need only render_to_string method because my templates have include inside.

Thanks!

**Comment by kraih** [comment, marked as answer]
I don't understand the purpose of that helper. It seems to be doing nothing. But you can access the application in Minion tasks via `$job->app`, so `my $output = $job->app->build_controller->render_to_string('word/some_prepared_xml', ...)` should just work without you having to do anything at all.

**Reply by saintserge** [reply]
I didn't knew about "build_controller" method, this is why I did it that way.
Thank you Sebastian!
